### PR TITLE
refactor: switch to a monorepo with lerna

### PR DESCRIPTION
**Pull request by kylelaker**
This refactors the repository into a more typical monorepo setup. This will eventually allow to add additional packages (perhaps an API client or a package that exports OSCAL types for TypeScript). Along with this refactoring, we've replaced `create-react-library` with `rollup` for the React library and we use `craco` instead of raw `react-scripts` in the Viewer app (to workaround some wonkiness with forbidden imports outside of `src/`).

Along with this we get TypeScript!! Only one or two files have been rewritten as part of this migration. That serves two purposes, first that we can use TypeScript, and second that we can adopt it incrementally.

We also have a `scripts/` directory which will allow us to store scripts that we want that are too complex to just toss into `package.json`. I would like to keep those scripts limited to Bash (all our users should have access to a basic Bash environment) or to TypeScript (we install `ts-node` as a dev dependency in the root repo).

`lerna` is used as the tool to manage our monorepo. A lot of features are now built into the `npm` workspaces features; however, `lerna` gives us some nice ability to manage references between repos and to run commands against all of them (and look at how nice that is for `scripts/watch-viewer`!)

The tests are all still `jest` and our linting rules remain largely the same. Tests should be run in the individual packages for now; functionally for the majority of our work, this means that you need to be in `packages/oscal-react-library` in order to run its tests for example. For commands that do work across all packages without any dependencies (for now, just `lint`) those can be run across all packages from the root.

If you do want to run across all packages, `lerna exec --stream` is a good friend; additionally, its `--scope` and `--include-dependencies` and `--include-dependents` arguments may be good friends.

You _do not_ need to install `lerna` globally (although you may find it useful to do so). You can use `npx lerna` (as the example in `scripts/watch-viewer` does).

Closes: #732 
Closes: #403
Closes: #596 
Closes: #493

### CODE REVIEW

**Comment by kylelaker** [comment]
Todo:

- [ ] Update documentation (CONTRIBUTING & README)
- [ ] Scripts for building/running work
- [ ] ESLint works
- [ ] Dependency update workflow works
- [ ] Basic TypeScript works (at least well enough for scripts & new packages)

**Review by tuckerzp — COMMENTED** [review]
This looks great so far and I am so excited to be able to start using Typescript! Without tests and linting working, I dont have too much to comment on. It looks good to me and I think the overall structure of things looks good. 

I do think there are some places in the `README` we should update to make using things more clear. It would also help in reviewing :smile:

**Comment by Bronstrom** [comment]
Looking over the current draft, I don't see anything that stands out as an issue at this point. The chosen direction and file structure seem like a logical choice. I'm liking a lot how `OSCALPropUtils.ts` is turning out.  Also, it's exciting were just on the cusp of our first TS implementation!

**Comment by kylelaker** [comment]
It built in the pipeline!!

**Comment by kylelaker** [comment]
There are some things that can't easily be tested here... GitHub Actions workflows that run on the `main`/`develop` branches are a big one. Additionally, other repos that depend on the structure of this (such as the PR that I created for the all-in-one container) may not show themselves until after this is merged.

Rather than trying to predict all those locations, I'd like to merge this before we have to deal with more conflicts; and then work quickly to find those other plcaes.

File: README.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ dependencies need to be installed initially with the command: `npm install`. Fro
 immediately run the application, use the following command:
 
 ```text
-npm run build-library-and-run-example
+./scripts/watch-viewer
 ```
 
 ### Testing
```
[the remark this answers is not in the document]
Comment (kylelaker) [reply README.md:41]: This is addressed now in the opening PR message.

**Review by tuckerzp — APPROVED** [review]
Thank you for your work on this @kylelaker! This is a super exciting change!!

**Review by Bronstrom — APPROVED** [review]
Looks good @kylelaker!

**Pull request merged**